### PR TITLE
CP-24602: Ported build from _oasis to jbuilder.

### PR DESCRIPTION
**Pull request by kc284**
Signed-off-by: Konstantina Chremmou <konstantina.chremmou@citrix.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+test:
+	jbuilder runtest
+
+# requires odoc
+doc:
+	jbuilder build @doc
+
+reindent:
+	git ls-files '*.ml*' '**/*.ml*' | xargs ocp-indent --syntax cstruct -i
```
Comment (mseri) [inline Makefile]: `'*.ml*'` is enough, ls-files looks for the files recursively in the folders

File: lib/jbuild [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    netdev
+    sockios_compat
+  ))
+  (c_names (
+    addr_stubs
+    bridge_stubs
+    link_stubs
+  ))
+  (libraries (
+    stdext
```
Comment (mseri) [inline lib/jbuild]: Please modify this to use the relevant `xapi-stdext-` libraries

File: xapi-netdev.opam [hunk 3]
```diff
@@ -1,23 +1,15 @@
...
-remove: [
-  ["ocamlfind" "remove" "netdev"]
-]
+build: [[ "jbuilder" "build" "-p" name "-j" jobs ]]
 available: [ os = "linux" ]
 depends: [
-  "ocamlfind"
+  "jbuilder" {build}
   "xapi-stdext"
   "xapi-forkexecd"
```
Comment (mseri) [inline xapi-netdev.opam]: `"forkexec"` should be enough for this

File: xapi-netdev.opam [hunk 4]
```diff
@@ -1,23 +1,15 @@
...
-  [make "install" "BINDIR=%{bin}%"]
-]
-remove: [
-  ["ocamlfind" "remove" "netdev"]
-]
+build: [[ "jbuilder" "build" "-p" name "-j" jobs ]]
 available: [ os = "linux" ]
 depends: [
-  "ocamlfind"
+  "jbuilder" {build}
```
Comment (mseri) [inline xapi-netdev.opam]: Please update the code to use the relevant `xapi-stdext-` libraries

**Comment by kc284** [comment]
This one too needs spec file update.

**Comment by mseri** [comment]
Do not merge until the spec file changes and the other PRs are ready for merge

**Pull request merged**